src/Microsoft.AspNet.Mvc.Core/FilterActionInvoker.cs

```diff
@@ -146,6 +152,149 @@ private IFilter[] GetFilters()
             }
         }
 
+        private async Task InvokeAllResourceFiltersAsync()
+        {
+            _cursor.SetStage(FilterStage.ResourceFilters);
+
+            var context = new ResourceExecutingContext(ActionContext, _filters);
+
+            context.InputFormatters = new List<IInputFormatter>(_inputFormatterProvider.InputFormatters);
+            context.InputFormatterSelector = _inputFormatterSelector;
```

Comment:
I'm not really sure what the design intent of input formatter selector is. Let's discuss when you're back.